src/Evolution/Systems/Cce/Equations.hpp

```diff
@@ -0,0 +1,1010 @@
+  }
+
+ private:
+  template <
+      typename TagsToSwshDifferentiateList, typename SwshDerivativeTagList,
+      typename IntegrandTagList, typename IndependentOfBondiIntegrationTagList,
+      typename TemporaryTagList, typename... TemporaryTagSubset,
+      typename... BondiAndSwCacheTagSubset, typename... SwshDerivativeTagSubset,
+      typename... IndependentOfBondiIntegrationTagSubset>
+  void dispatch_to_compute_from_scalars(
+      const gsl::not_null<Variables<IntegrandTagList>*> bondi_integrand_terms,
+      const gsl::not_null<Variables<TemporaryTagList>*> temporary_cache,
+      const Variables<TagsToSwshDifferentiateList>&
+          pre_swsh_derivative_bondi_quantities,
+      const Variables<SwshDerivativeTagList>& swsh_derivatives,
+      const Variables<IndependentOfBondiIntegrationTagList>&
+          values_independent_of_bondi_integration,
+      tmpl::list<TemporaryTagSubset...> /*meta*/,
+      tmpl::list<BondiAndSwCacheTagSubset...> /*meta*/,
+      tmpl::list<SwshDerivativeTagSubset...> /*meta*/,
+      tmpl::list<IndependentOfBondiIntegrationTagSubset...> /*meta*/) noexcept {
+    ComputeBondiIntegrand<IntegrandTag>::apply(
+        make_not_null(&get(get<IntegrandTag>(*bondi_integrand_terms))),
+        make_not_null(&get(get<TemporaryTagSubset>(*temporary_cache)))...,
+        get(get<BondiAndSwCacheTagSubset>(
+            pre_swsh_derivative_bondi_quantities))...,
+        get(get<SwshDerivativeTagSubset>(swsh_derivatives))...,
+        get(get<IndependentOfBondiIntegrationTagSubset>(
+            values_independent_of_bondi_integration))...);
+  }
+};
+
+/*!
+ * \brief Computes the integrand (right-hand side) of the equation which
+ * determines the radial (y) dependence of the Bondi quantity \f$\beta\f$.
+ *
+ * \details The quantity \f$\beta\f$ is defined via the Bondi form of the
+ * metric:
+ * \f[
+ * ds^2 = - \left(e^{2 \beta} (1 + r W) - r^2 h_{AB} U^A U^B\right) du^2 - 2
+ * e^{2 \beta} du dr - 2 r^2 h_{AB} U^B du dx^A + r^2 h_{A B} dx^A dx^B. \f]
+ * Additional quantities \f$J\f$ and \f$K\f$ are defined using a spherical
+ * angular dyad \f$q^A\f$:
+ * \f[ J \equiv h_{A B} q^A q^B, K \equiv h_{A B} q^A
+ * \bar{q}^B.\f]
+ * See \cite Bishop1997ik \cite Handmer2014qha for full details.
+ *
+ * in the compactified coordinate \f$ y \equiv 1 - 2 R/ r\f$, where \f$R\f$
+ * is determined by the worldtube initial data, the equation which
+ * determines \f$\beta\f$ on a surface of constant \f$u\f$ given \f$J\f$ on
+ * the same surface is, \f[\partial_y (\beta) = \frac{1}{8} (-1 + y)
+ * \left(\partial_y (J) \partial_y
+ * (\bar{J}) -  \frac{(\partial_y (J \bar{J}))^2}{4 K^2}\right). \f]
+ */
+template <>
+struct ComputeBondiIntegrand<Tags::Integrand<Tags::Beta>> {
+  // clang-tidy readability-avoid-const-params-in-decls, but the consts must
+  // stay
+  static void apply(
+      const gsl::not_null<SpinWeighted<ComplexDataVector, 0>*>  // NOLINT
```

Comment:
I think you can remove the `const`s here, you only need them in the `cpp`.

Reply:
That is not true for this scenario, because the calling interface in this case needs to know that it is passing to a function with const arguments (otherwise it can't rely on the function it calls to not mutate its own const arguments). At any rate, I wanted to avoid the long list of nolint's and it does not compile without the const :P

Reply:
Modulo a compiler bug this is not how `const` works. You can safely remove the `const` on the `not_null` arguments in the forward declarations, there's no need to `NOLINT` anything. On the arguments that are references, the `const` matters. I tested this that it compiles the `Cce` and `Test_Cce` libraries clang7 and gcc6. We also do not `const` the `not_null` arguments everywhere else in the entire repository and have never had to `NOLINT` anything. We actually use `not_null` so we don't have to `NOLINT`.

Reply:
Fortunately, this also can be reverted to standard use now that the `Variables` interface has been dropped.